src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsmanageprisonvisitsorchestration/utils/PublicBookerValidationUtil.kt

```diff
@@ -0,0 +1,41 @@
+package uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.utils
+
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.prisoner.search.PrisonerDto
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.visit.scheduler.VisitSchedulerPrisonDto
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.dto.visit.scheduler.enums.UserType
+import uk.gov.justice.digital.hmpps.hmppsmanageprisonvisitsorchestration.service.PrisonService
+
+@Component
+class PublicBookerValidationUtil {
+  companion object {
+    const val PRISONER_VALIDATION_ERROR_MSG = "prisoner validation for prisoner number - {0} failed with error - {1}"
+    const val PRISON_VALIDATION_ERROR_MSG =
+      "prison validation for prison code - {0} for prisoner number - {1} failed with error - {2}"
+  }
+
+  fun validatePrison(prison: VisitSchedulerPrisonDto, prisonService: PrisonService): String? {
```

Comment:
Instead of passing in a PrisonService, could we have it as a component of this class ?

class PublicBookerValidationUtil(   private val prisonService: PrisonService ) {}

and then remove from function

fun validatePrison(prison: VisitSchedulerPrisonDto): String? { }

Reply:
Updated